File: C/1 Arrays/Q7.c
// The program below takes a specified number of integer inputs, stores them in an array, and then determines and prints the highest and lowest values among the elements.


#include <stdio.h>

int main()
{
    int arr1[400];
    int i, mx, mn, n;

    printf("Enter the number of elements to be stored in the array :");
    scanf("%d", &n);

    printf("Input %d elements in the array :\n", n);
    for (i = 0; i < n; i++) {
        printf("element - %d : ", i + 1);
        scanf("%d", &arr1[i]);
    }

    mx = arr1[0];
    mn = arr1[0];

    for (i = 1; i < n; i++) {
        if (arr1[i] > mx)
            mx = arr1[i];

        if (arr1[i] < mn)
            mn = arr1[i];
    }

    printf("Maximum element is : %d\n", mx);
    printf("Minimum element is : %d\n\n", mn);
	return 0;
}

File: Rust/1 Arrays/Q7.rs
// The program below takes a specified number of integer inputs, stores them in an array, and then determines and prints the highest and lowest values among the elements.
// Author: Morteza Farrokhnejad

use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter the number of elements:");
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let n: usize = input.trim().parse().expect("Please enter a valid number");
    input.clear();

    if n == 0 {
        println!("No elements to process.");
        return;
    }

    let mut numbers = Vec::new();
    println!("Enter the elements of the array:");
    for i in 0..n {
        println!("element - {}:", i + 1);
        io::stdin().read_line(&mut input).expect("Failed to read input");
        let element: i32 = input.trim().parse().expect("Please enter a valid integer");
        numbers.push(element);
        input.clear();
    }

    let min = numbers.iter().min().unwrap();
    let max = numbers.iter().max().unwrap();

    println!("\nThe highest value is: {}", max);
    println!("The lowest value is: {}", min);
}
